TestCntlrApp/src/enbApp/nb_dam.c

```diff
@@ -705,65 +888,122 @@ U8                           msgType
  *    -#Success : ROK
  *    -#Failure : RFAILED
  */
-PUBLIC S16 nbDamPcapDatInd
-(
- Buffer                       *mBuf
-)
+PUBLIC S16 nbDamPcapDatInd(Buffer *mBuf)
 {
-   U32                       ueIpAddr;
-   MsgLen                    len = 0;
-   NbDamTnlCb                *tnl;
-   NbDamUeCb                 *ueCb = NULLP;
-   U8  ipPkt[4] = {0}; 
-   U8  idx = 0; 
-   U8 drbId; 
-   MsgLen ipIdx = 12;
-
-   EgtUEvnt                  *eguEvtMsg;
-   EgUMsg                    *egMsg;
-   NbDamDrbCb                *drbCb;
-
-   SFndLenMsg(mBuf, &len);
-   if(len == 0)
-   {
-      NB_LOG_ERROR(&nbCb,"Empty Buffer recieved");
+  MsgLen len = 0;
+  NbDamTnlCb *tnl;
+  NbDamUeCb *ueCb = NULLP;
+  U8 ipPkt[NB_PACKET_SIZE] = {0};
+  U8 idx = 0;
+  U8 drbId;
+  MsgLen ipIdx = 1; // Start from 1st index-DSCP
+  NbIpPktFields ipPktFields = {0};
+
+  EgtUEvnt *eguEvtMsg;
+  EgUMsg *egMsg;
+  NbDamDrbCb *drbCb;
+
+  SFndLenMsg(mBuf, &len);
+  if (len == 0) {
+    NB_LOG_ERROR(&nbCb, "Empty Buffer recieved");
+    SPutMsg(mBuf);
+    RETVALUE(ROK);
+  }
+
+  /* Fetch ToS or DSCP*/
+  for (idx = 0; idx < 1; idx++) {
+    if ((SExamMsg(&ipPkt[idx], mBuf, ipIdx) != ROK)) {
+      NB_LOG_ERROR(&nbCb, "Failed to fetch ToS");
       SPutMsg(mBuf);
-      RETVALUE(ROK);
-   }
-
-   for(idx = 0; idx < 4; idx++)
-   {
-      if((SExamMsg(&ipPkt[idx], mBuf, ipIdx) != ROK))
-      {    
-         NB_LOG_ERROR(&nbCb,"Failed to get message type");
-         SPutMsg(mBuf);
-         RETVALUE(RFAILED);
-      }
-      ipIdx++;
-   }
-   ueIpAddr = (ipPkt[0] << 24) + (ipPkt[1] << 16) +
-      (ipPkt[2] << 8 ) + ipPkt[3];
-
-   /* get the ueCb */
-   ueCb = nbDamGetueCbkeyUeIp(ueIpAddr, &drbId);
-   if(ueCb == NULLP)
-   {
-      NB_FREEMBUF(mBuf);
       RETVALUE(RFAILED);
-   }
-
-   /* mark for ue data received */
-   ueCb->dataRcvd = TRUE;   
-   /* get the tunnel Cb */
-   if ( ROK != (cmHashListFind(&(ueCb->drbs), (U8 *)&(drbId),
-               sizeof(U8), 0, (PTR *)&drbCb)))
-   {    
+    }
+    ipIdx++;
+  }
+  /* Tos or DSCP*/
+  ipPktFields.srvClass = ipPkt[0];
+  /* Skip 9 bytes to fetch the protocol ID*/
+  ipIdx = 9;
+  /* Fetch protocol Id*/
+  for (idx = 0; idx < 1; idx++) {
```

Comment:
again why for loop if iterated only once?

Reply:
done